packages/react/src/firestore/useGetCountFromServerQuery.test.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { type ReactNode } from "react";
+import { describe, expect, test, beforeEach } from "vitest";
+import { useGetCountFromServerQuery } from "./useGetCountFromServerQuery";
+import { renderHook, waitFor } from "@testing-library/react";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { collection, addDoc, query, where } from "firebase/firestore";
+import {
+  expectFirestoreError,
+  firestore,
+  wipeFirestore,
+} from "~/testing-utils";
+
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      retry: false,
+    },
+  },
+});
+
+const wrapper = ({ children }: { children: ReactNode }) => (
+  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+);
+
+describe("useGetCountFromServerQuery", () => {
+  beforeEach(async () => await wipeFirestore());
+
+  test("returns correct count for empty collection", async () => {
+    const collectionRef = collection(firestore, "tests");
+
+    const { result } = renderHook(
+      () =>
+        useGetCountFromServerQuery(collectionRef, {
+          queryKey: ["count", "empty"],
+        }),
+      { wrapper }
+    );
+
+    await waitFor(() => expect(result.current.isSuccess).toBe(true));
+
+    expect(result.current.data?.data().count).toBe(0);
+  });
+
+  test("returns correct count for non-empty collection", async () => {
+    const collectionRef = collection(firestore, "tests");
+
+    await addDoc(collectionRef, { foo: "bar1" });
+    await addDoc(collectionRef, { foo: "bar2" });
+    await addDoc(collectionRef, { foo: "bar3" });
+
+    const { result } = renderHook(
+      () =>
+        useGetCountFromServerQuery(collectionRef, {
+          queryKey: ["count", "non-empty"],
+        }),
+      { wrapper }
+    );
+
+    await waitFor(() => expect(result.current.isSuccess).toBe(true));
+
+    expect(result.current.data?.data().count).toBe(3);
```

Comment:
there is optional chaining at `data?.data()` because [useQuery](https://tanstack.com/query/latest/docs/framework/react/reference/useQuery#usequery)  initializes with undefined data before the query resolves
<img width="885" alt="image" src="https://github.com/user-attachments/assets/6c7731dd-10ad-4a85-bf32-f201fefb8c40">

Reply:
That's OK, it's because typescript isn't inferring isSuccess with data (since it's inside of waitFor).